go_annotation/fairseq_layers/esm_model.py

```diff
@@ -0,0 +1,133 @@
+import torch
+from fairseq.data import Dictionary
+from fairseq.models import FairseqEncoder, register_model, register_model_architecture
+from fairseq.models.roberta import RobertaModel, utils
+from torch import nn
+
+
+@register_model("esm")
+class ESMModel(RobertaModel):
+    def __init__(self, args, encoder):
+        super().__init__(args, encoder)
+        self.args = args
+        self.classification_heads = nn.ModuleDict()
+
+    @staticmethod
+    def add_args(parser):
+        """Add model-specific arguments to the parser."""
+        parser.add_argument(
+            "--esm-architecture", type=str, help="ESM pretrained architecture"
+        )
+        parser.add_argument(
+            "--max-positions", type=int, help="number of positional embeddings to learn"
+        )
+
+        parser.add_argument(
+            "--pooler-activation-fn",
+            choices=utils.get_available_activation_fns(),
+            help="activation function to use for pooler layer",
+        )
+        parser.add_argument(
+            "--pooler-dropout",
+            type=float,
+            metavar="D",
+            help="dropout probability in the masked_lm pooler layers",
+        )
+        # args for Training with Quantization Noise for Extreme Model Compression ({Fan*, Stock*} et al., 2020)
+        parser.add_argument(
+            "--quant-noise-pq",
+            type=float,
+            metavar="D",
+            default=0,
+            help="iterative PQ quantization noise at training time",
+        )
+        parser.add_argument(
+            "--quant-noise-pq-block-size",
+            type=int,
+            metavar="D",
+            default=8,
+            help="block size of quantization noise at training time",
+        )
+        parser.add_argument(
+            "--quant-noise-scalar",
+            type=float,
+            metavar="D",
+            default=0,
+            help="scalar quantization noise and scalar quantization at training time",
+        )
+        # args for "Better Fine-Tuning by Reducing Representational Collapse" (Aghajanyan et al. 2020)
+        parser.add_argument(
+            "--spectral-norm-classification-head",
+            action="store_true",
+            default=False,
+            help="Apply spectral normalization on the classification head",
+        )
+
+    @classmethod
+    def build_model(cls, args, task):
+
+        print(args.esm_architecture)
+        model, alphabet = torch.hub.load("facebookresearch/esm", args.esm_architecture)  # fix
```

Comment:
I believe models are set to `eval` mode by default after loading from torch hub. Do we need to set `model.train()` here? Or is it possibly set within fairseq somewhere?

Reply:
good catch, we should probably throw in a model.train() somewhere then.